src/basic/DAXPY-Sycl.cpp

```diff
@@ -0,0 +1,134 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-19, Lawrence Livermore National Security, LLC.
+//
+// Produced at the Lawrence Livermore National Laboratory
+//
+// LLNL-CODE-738930
+//
+// All rights reserved.
+//
+// This file is part of the RAJA Performance Suite.
+//
+// For details about use and distribution, please read RAJAPerf/LICENSE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "DAXPY.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#if defined(RAJA_ENABLE_SYCL)
+
+#include "common/SyclDataUtils.hpp"
+
+#include <iostream>
+
+namespace rajaperf
+{
+namespace basic
+{
+
+#define DAXPY_DATA_SETUP_SYCL \
+  allocAndInitSyclDeviceData(x, m_x, iend, qu); \
```

Comment:
We shouldn't have to do this if we add sycl allocation support to KernelBase.

Reply:
Right. That's one of the things I was referring to with my "align" comment above. Do you want to start a checklist above for items we need to work on?